### PR TITLE
Add developer translation documentation

### PR DESCRIPTION
**Pull request by Kissaki**

### CODE REVIEW

**Comment by Kissaki** [comment]
This is missing Weblate workflow and conflict+lock resolution documentation. I created #150 for that (can be done after).

**Review by Krzmbrzl — APPROVED** [review]

**Comment by Krzmbrzl** [comment]
The resource link formatting commit should probably be squashed with the first commit before merge :point_up:

File: hugo/content/documentation/translation.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+---
+title: Developer Translation Documentation
+date: 2021-03-14
+lastmod: 2021-03-14
+---
+Mumble makes use of the Qt framework. This is also the case for translations. We follow the Qt tools and translation workflow.
+
+Translators can translate our project in our [Mumble Weblate project](https://hosted.weblate.org/projects/mumble/mumble-client/) without a need for Git, Code, or other programs than their webbrowser.
```
Comment (davidebeatrici) [inline hugo/content/documentation/translation.md:8]:
```suggestion
Translators can translate our project in our [Mumble Weblate project](https://hosted.weblate.org/projects/mumble/mumble-client/) without a need for Git, Code, or programs other than their web browser.
```

**Review by davidebeatrici — APPROVED** [review]

**Comment by Kissaki** [comment]
> 
> 
> The resource link formatting commit should probably be squashed with the first commit before merge ☝️

Yes, I planned to do that. I pushed it separate in case you were already looking at the changes, reviewing them.

**Pull request merged**